### PR TITLE
Add when to generalize section to development doc

### PR DESCRIPTION
**Pull request by idelahoz**

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -63,6 +63,28 @@ When working on a feature, the order you work on things matter. Think of a featu
...
+-  **The extra code you write may not be used at all:**
+    The other scenarios where the code piece will be needed may never come, but you may have spent 4 more hours trying to make your code work for all future scenarios that will not exist.
+
+- **You may lose a lot more time on a task that shouldnt normally take that long:**
+    Yeah, of course this is ususally answered with the "I'm saving a lot of time for later" response, but you never know the actual amount of time you're saving for the future. If you need to do something ASAP, dont waste time today to save you some unknown time tomorrow.
+
+- **You may not be aware of what things need to be common between scenarios:**
+    Writing code for one scenario may not give you enough information about the common things among all the future scenarios. Usually the magic number of cases is 3, after you copy and modify a piece of code for the third time you should know in that moment what needs to be in all scenarios and extract that in a class or any other extensible code statement.
+
+#### when to generalize:
```
Comment (venkatd) [inline DEVELOPMENT.md]: When

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -63,6 +63,28 @@ When working on a feature, the order you work on things matter. Think of a featu
 - Minor Annoyances: Things that hurt the user experience but the person can get over. Example would be an error message popping up in the middle but if they hit ok they can keep going
 - Minor Refactoring
 
+### Should I make my code more general?
+Sometimes we consider adding more flexibility to our current code in order to make future changes easier or to make a piece of code usable for some slightly different scenarios, this process has a couple issues:
+
+-  **The extra code you write may not be used at all:**
```
Comment (venkatd) [inline DEVELOPMENT.md]: These headings have ":" but the rest of the document doesn't. We need consistency.

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -63,6 +63,28 @@ When working on a feature, the order you work on things matter. Think of a featu
...
+    Yeah, of course this is ususally answered with the "I'm saving a lot of time for later" response, but you never know the actual amount of time you're saving for the future. If you need to do something ASAP, dont waste time today to save you some unknown time tomorrow.
+
+- **You may not be aware of what things need to be common between scenarios:**
+    Writing code for one scenario may not give you enough information about the common things among all the future scenarios. Usually the magic number of cases is 3, after you copy and modify a piece of code for the third time you should know in that moment what needs to be in all scenarios and extract that in a class or any other extensible code statement.
+
+#### when to generalize:
+
+The 3 cases rule mentioned before is very useful for this, if you have more than 2 cases where a piece of code is repeated, you should have a pretty accurate idea of what needs to be generalized and how to do it.
+
+There are some cases when generalizing right away may be worth it. there's a couple questions you can ask yourself to identify them:
```
Comment (venkatd) [inline DEVELOPMENT.md]: There's

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -63,6 +63,28 @@ When working on a feature, the order you work on things matter. Think of a featu
...
+
+- How much work will take to generalize this code now?
```
Comment (venkatd) [inline DEVELOPMENT.md]: How much work will it be to generalize this code now?

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -63,6 +63,28 @@ When working on a feature, the order you work on things matter. Think of a featu
...
+- How much work will take to generalize this code when I need it to handle other cases?
+
+If the answer to the second questions is a lot more than the answer to the first one, it may be a a good idea to think about generalizing right away.
```
Comment (venkatd) [inline DEVELOPMENT.md]: "second question" -- there is only one question

Comment (venkatd) [inline DEVELOPMENT.md]: I would rephrase this sentence so it's not pointing somewhere else. Like so:

> If generalizing later will increase the overall work by a lot, then you should generalize it right away.

File: DEVELOPMENT.md [hunk 6]
```diff
@@ -63,6 +63,28 @@ When working on a feature, the order you work on things matter. Think of a featu
 - Minor Annoyances: Things that hurt the user experience but the person can get over. Example would be an error message popping up in the middle but if they hit ok they can keep going
 - Minor Refactoring
 
+### Should I make my code more general?
+Sometimes we consider adding more flexibility to our current code in order to make future changes easier or to make a piece of code usable for some slightly different scenarios, this process has a couple issues:
```
Comment (venkatd) [inline DEVELOPMENT.md]: This process -- new sentence

Comment (idelahoz) [inline DEVELOPMENT.md]: Ahh, so it needs a period.

**Comment by idelahoz** [comment]
@venkatd I updated this last week. take a look when you get a chance.
